File: thirdSeminar/src/main/java/org/sopt/thirdSeminar/common/dto/ErrorResponse.java
package org.sopt.thirdSeminar.common.dto;

import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.ToString;
import org.sopt.thirdSeminar.exception.ErrorMessage;

@Getter
@ToString
@RequiredArgsConstructor
public class ErrorResponse {
    private final int code;
    private final String message;

    public static ErrorResponse error(ErrorMessage error) {
        return new ErrorResponse(error.getStatus().value(), error.getMessage());
    }


    public static ErrorResponse error(ErrorMessage error, String message) {
        return new ErrorResponse(error.getStatus().value(), message);
    }
}

File: sixthKotlin/src/main/kotlin/org/sopt/sixthKotlin/common/dto/ErrorResponse.kt
package org.sopt.sixthKotlin.common.dto

import org.sopt.sixthKotlin.exception.ErrorMessage

class ErrorResponse(
    private val code: Int,
    private val message: String
) {

    companion object {
        fun error(error: ErrorMessage): ErrorResponse {
            return ErrorResponse(error.status.value(), error.message)
        }

        fun error(error: ErrorMessage, message: String): ErrorResponse {
            return ErrorResponse(error.status.value(), message)
        }
    }
}